### ISSUE TITLE
Class ClientInfo: Incorrect Agent detected

### ISSUE THREAD

**Issue by Atrix1987** [restlet/restlet-framework-java#1238]
For: `restlet-gae-2.3.7`.

``` java
Product prod = req.getClientInfo().getMainAgentProduct();
```

The above code returns incorrect User Agent/Product Name. It returns a Product with 
name = **"Version"**
version= **"9.1.3"**
whereas the URL was opened from a Safari browser with version 9.1.3

When I try the same URL from Opera, it returns Chrome as the detected Agent.

PFB all relevant headers:

``` json
{
    "attributes": {
        "org.restlet.http.version": "1.1",
        "org.restlet.http.headers": [{
            "name": "Host",
            "value": "xxx.appspot.com"
        }, {
            "name": "User-Agent",
            "value": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/601.7.8 (KHTML, like Gecko) Version/9.1.3 Safari/601.7.8"
        }, {
            "name": "Accept",
            "value": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"
        }, {
            "name": "Accept-Language",
            "value": "en-us"
        }, {
            "name": "X-AppEngine-Country",
            "value": "IN"
        }, {
            "name": "X-AppEngine-Region",
            "value": "ka"
        }, {
            "name": "X-AppEngine-City",
            "value": "bangalore"
        }, {
            "name": "X-AppEngine-CityLatLong",
            "value": "12.971599,77.594563"
        }, {
            "name": "X-Cloud-Trace-Context",
            "value": "8c382f1be13e81xxxxx4fa3a133792868xxx3454;o=1"
        }],
        "com.google.apphosting.runtime.jetty.APP_VERSION_REQUEST_ATTR": {},
        "org.restlet.startTime": 1473246222156
    },
    "getAgentName": "Version",
    "agentVersion": "9.1.3",
    "agentAttributes": {
        "agentOsShortName": "Macintosh",
        "osData": "Intel Mac OS X 10_11_6",
        "agentName": "Version",
        "safariVersion": "601.7.8",
        "mozillaVersion": "5.0",
        "appleWebKitComment": "KHTML, like Gecko",
        "agentVersion": "9.1.3",
        "appleWebKitVersion": "601.7.8"
    },
    "products": [{
        "name": "Mozilla",
        "comment": "Macintosh; Intel Mac OS X 10_11_6",
        "version": "5.0"
    }, {
        "name": "AppleWebKit",
        "comment": "KHTML, like Gecko",
        "version": "601.7.8"
    }, {
        "name": "Version",
        "version": "9.1.3"
    }, {
        "name": "Safari",
        "version": "601.7.8"
    }],
    "mainAgent": {
        "name": "Version",
        "version": "9.1.3"
    },
    "getAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/601.7.8 (KHTML, like Gecko) Version/9.1.3 Safari/601.7.8"
}
```

**Comment by thboileau** [comment]
Thanks for pointing this issue.
The fix is available in branch 2.3 and will be shipped with version 2.3.8.

**Comment by ArjohnKampman** [comment]
This fix introduced a new issue due to a typo in the agent.properties file. This file now contains "{a,gentOs}" for Firefox for Windows; where the comma was likely added by accident. As a result, all requests to the server now log a warning "An invalid character was detected inside a pattern variable : null".

**Comment by chrbongardt** [comment]
Hi, I can also confirm this issue. We're getting a warning in on line 543 of Template

**Comment by thboileau** [comment]
thanks for all your messages.
The fix will be part of version 2.3.10 that will be released soon.

**Comment by thboileau** [comment]
hi  @ArjohnKampman , @chrbongardt 
Version 2.3.10 has been released with a fix. Could you confirm it works for you?

I feel sorry for the mistake.
